sources/api/sundog/src/main.rs

```diff
@@ -221,6 +224,54 @@ where
     Ok(populated_settings)
 }
 
+// Builds the proxy environment variables to pass to settings generators
+async fn build_proxy_env<P>(socket_path: P) -> Result<HashMap<String, String>>
+where
+    P: AsRef<Path>,
+{
+    // Retrieve network proxy-related settings.
+    let settings: serde_json::Value =
+        schnauzer::get_json(&socket_path, "/settings", Some(("prefix", "network")))
+            .await
+            .context(error::RetrieveSettingsSnafu)?;
```

Comment:
nit: using `schnauzer` is inconsistent with the other API accesses in this code. I'd prefer all one way or the other, but I'm OK with a backlog item for converting over the rest if that's what we want the idiom to be going forward.